error.go

```diff
@@ -124,6 +124,33 @@ func (e *Err) Error() string {
 	return fmt.Sprintf("%s: %v", e.message, err)
 }
 
+// Format implements fmt.Formatter
+// When printing errors with %+v it also prints the stack trace.
+// %#v unsurprisingly will print the real underlying type.
+func (e *Err) Format(s fmt.State, verb rune) {
+	switch verb {
+	case 'v':
+		switch {
+		case s.Flag('+'):
+			fmt.Fprintf(s, "%s", ErrorStack(e))
+			return
+		case s.Flag('#'):
+			// avoid infinite recursion by wrapping e into a type
+			// that doesn't implement Formatter.
+			fmt.Fprintf(s, "%#v", (*unformatter)(e))
+			return
+		}
+		fallthrough
+	case 's':
+		fmt.Fprintf(s, "%s", e.Error())
+	}
+}
+
+// helper for Format
+type unformatter Err
+
+func (unformatter) Format() { /* break the fmt.Formatter interface */ }
```

Comment:
well, in that case it would fulfill the `fmt.Formatter` interface.
By adding a method with an incompatible type signature we're breaking the interface.

This allows us to do:
```
fmt.Fprintf(s, "%#v", (*unformatter)(e))
```

otherwise `fmt.Fprintf` will do a runtime type assertion `e.(fmt.Formatter)`, which will succeed and cause infinite recursion here.

@mjs, do you know of a better  way to preserve the default rendering that the object would have had with `%#v`?